'''
Handle the exception thrown by the code below by using try and except blocks.

In [1]:
for i in ['a','b','c']:
    print(i**2)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-c35f41ad7311> in <module>()
      1 for i in ['a','b','c']:
----> 2     print(i**2)

TypeError: unsupported operand type(s) for ** or pow(): 'str' and 'int'
'''


def hw1():
    try:
        for i in ['a', 'b', 'c']:
            print(i**2)
    except TypeError:
        print("list of string can't be squared by 2")

hw1()

'''
Problem 2
Handle the exception thrown by the code below by using try and except blocks. Then use a finally block to print 'All Done.'

In [2]:
x = 5
y = 0

z = x/y
---------------------------------------------------------------------------
ZeroDivisionError                         Traceback (most recent call last)
<ipython-input-2-6f985c4c80dd> in <module>()
      2 y = 0
      3 
----> 4 z = x/y

ZeroDivisionError: division by zero
'''


def hw2():
    try:
        x = 5
        y = 0
        z = x / y
    except ZeroDivisionError:
        print("Can't division by zero")


hw2()


'''
Write a function that asks for an integer and prints the square of it. 
Use a while loop with a try, except, else block to account for incorrect inputs.

In [3]:
def ask():
    pass
In [4]:
ask()
Input an integer: null
An error occurred! Please try again!
Input an integer: 2
Thank you, your number squared is:  4
'''


def hw3():
    while True:
        try:
            result = input("Please enter an integer: ")
            result = int(result) ** 2
        except:
            print("An error occurred! Please try again!")
            continue
        else:
            print("Thank you, your number squared is: {}".format(result))
            break
        finally:
            print("End of try")

hw3()